autopush/websocket.py

```diff
@@ -550,6 +583,25 @@ def _send_ping(self):
         self.metrics.increment("updates.client.ping", tags=self.base_tags)
         return self.sendMessage("{}", False)
 
+    def set_active(self):
+        self.idle = ms_time()
+
+    def check_idle(self):
+        if self.idler is not None:
+            # cancel old idler
+            self.idler.cancel()
+        try:
+            if self.udp is not None:
+                if ms_time() - self.idle >= self.udp_timeout * 1000:
+                    self.sendClose(code=4774, reason='UDP Wakeup')
+                    return
+                self.idler = self.deferToLater(self.udp_timeout,
+                                               self.check_idle)
+        except (KeyError, AttributeError):
+            # More than likely, this isn't a UDP wake connection.
+            # log.msg("Warn: could not check idle %s" % x)
+            pass
```

Comment:
Should probably be a method on the implementation of IWakeup that indicates whether an 'idle' connection should be closed, and how.